### ISSUE TITLE
关于GRPO训练采样重复的问题

### ISSUE THREAD

**Issue by munian08** [modelscope/ms-swift#3450]
脚本我是仿照多模态GRPO完整实验流程写的，在8卡GPU上实验，2卡部署，6卡训练。num_generations设置的是6，我理解的就是每个prompt采样6次，但是通过wandb中打印的completion中发现6个输出中有3个是重复的，想问一下大佬这是我哪里设置有误吗

ms-swift 3.2.0
vllm 0.7.3

<img width="242" alt="Image" src="https://github.com/user-attachments/assets/6a393ccd-708d-4be5-8a14-77744651a97a" />

```
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=6 \
MAX_PIXELS=150528 \
swift rlhf \
    --rlhf_type grpo \
    --model /qwen2vl-7b-instruct \
    --model_type qwen2_vl \
    --lazy_tokenize True \
    --external_plugins /rl_plugin.py \
    --reward_funcs format acc_num \
    --use_vllm true \
    --vllm_device auto \
    --vllm_gpu_memory_utilization 0.6 \
    --vllm_max_model_len 8192 \
    --vllm_limit_mm_per_prompt '{"image": 3}' \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear lm_head \
    --torch_dtype bfloat16 \
    --dataset '/cleaned.jsonl' \
    --max_length 8192 \
    --max_completion_length 1024 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 8 \
    --eval_steps 500 \
    --save_steps 500 \
    --save_total_limit 5 \
    --logging_steps 5 \
    --output_dir output/qwen2vl-7b \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 6 \
    --temperature 1.2 \
    --system '' \
    --deepspeed zero3 \
    --log_completions true \
    --report_to wandb \
    --num_iterations 1 \
    --num_infer_workers 2 \
    --beta 0
  
```

**Comment by munian08** [comment]
> It should be fixed in [#3437](https://github.com/modelscope/ms-swift/pull/3437).

我拉了最新的代码，在sh中加上--seed 42，这样好像就没有对同一条message多次采样？打印出来的{"step": ["0", "0", "0", "0", "0", "0"]对应的messages是不同的，可以麻烦再看一下嘛

**Comment by hjh0119** [comment]
复现了 应该是个bug

**Comment by Peng-YM** [comment]
最近版本看上去还有同样的问题，可能因为 seed 的逻辑又回归了

**Comment by tastelikefeet** [comment]
我试一下

**Comment by tastelikefeet** [comment]
![Image](https://github.com/user-attachments/assets/e5d79b2b-315e-4913-af7b-20d5c5617bab)
我用最新的main分支代码是不复现的，遇到问题的同学可以用最新的main分支代码再试一下吗

**Comment by BC-A** [comment]
If num_return_sequences is greater than 2, such as 10, there will still be a problem of high repetitiveness. The latest main code is used

**Comment by hjh0119** [comment]
@BC-A any repro script?
